### PR TITLE
Bazel improvements

### PR DESCRIPTION
**Pull request by mehrdadn**
## Why are these changes needed?

- Bazel environment has too many variables and this hurts reproducibility and caching across systems

- Bazel outputs are too noisy and can make it difficult when viewing the log

- When stack traces aren't shown on Linux, `strace` can be useful

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

File: ci/travis/install-bazel.sh [hunk 1]
```diff
@@ -38,14 +38,21 @@ else
...
+build --curses=no
+# Use bazel disk cache if this script is running in Travis.
+build --disk_cache="${HOME}/ray-bazel-cache"
+build --progress_report_interval=60
+# Use ray google cloud cache
+build --remote_cache="https://storage.googleapis.com/ray-bazel-cache"
+build --show_progress_rate_limit=15
+build --show_timestamps
+build --ui_actions_shown=1024
+build --watchfs
```
Comment (simon-mo) [inline ci/travis/install-bazel.sh]: maybe take this out for CI env?

File: .bazelrc [hunk 2]
```diff
@@ -1,13 +1,10 @@
 # Must be first. Enables build:windows, build:linux, build:macos, build:freebsd, build:openbsd
 build --enable_platform_specific_config
 ###############################################################################
-build --action_env=BAZEL_LLVM
-build --action_env=BAZEL_SH
+# On       Windows, provide: USE_CLANG_CL=1, CC=clang, BAZEL_LLVM, BAZEL_SH
+# On all platforms, provide: PYTHON2_BIN_PATH, PYTHON3_BIN_PATH
```
Comment (simon-mo) [inline .bazelrc]: I'll test this on mac

File: ci/travis/install-bazel.sh [hunk 3]
```diff
@@ -38,14 +38,21 @@ else
...
-fi
 if [ "${TRAVIS-}" = true ] || [ -n "${GITHUB_TOKEN-}" ]; then
-  # Use ray google cloud cache
-  echo "build --remote_cache=https://storage.googleapis.com/ray-bazel-cache" >> "${HOME}/.bazelrc"
+  mkdir -p "${HOME}/ray-bazel-cache"
+  cat <<EOF >> "${HOME}/.bazelrc"
+# CI output doesn't scroll, so don't use curses
+build --curses=no
+# Use bazel disk cache if this script is running in Travis.
+build --disk_cache="${HOME}/ray-bazel-cache"
```
Comment (simon-mo) [inline ci/travis/install-bazel.sh]: Take a look at github action warnings and try to fix them

File: ci/travis/install-dependencies.sh [hunk 4]
```diff
@@ -32,14 +32,28 @@ fi
...
 if [[ "$PYTHON" == "3.6" ]] && [[ "$platform" == "linux" ]]; then
   sudo apt-get update
-  sudo apt-get install -y python-dev python-numpy build-essential curl unzip tmux gdb
+  sudo apt-get install -y python-dev python-numpy build-essential curl unzip tmux gdb libunwind-dev
   # Install miniconda.
   wget -q https://repo.continuum.io/miniconda/Miniconda3-4.5.4-Linux-x86_64.sh -O miniconda.sh -nv
   bash miniconda.sh -b -p $HOME/miniconda
   export PATH="$HOME/miniconda/bin:$PATH"
-  pip install -q scipy tensorflow==$tf_version \
+  case "$(lsb_release -r -s || true)" in
```
Comment (simon-mo) [inline ci/travis/install-dependencies.sh]: can you move this to `ci/travis/install-strace.sh` ?

File: ci/travis/install-dependencies.sh [hunk 5]
```diff
@@ -32,14 +32,28 @@ fi
 # Upgrade pip and other packages to avoid incompatibility ERRORS.
 pip install --upgrade pip # setuptools cloudpickle urllib3
 
+# If we're in a CI environment, do some configuration
+if [ "${TRAVIS-}" = true ] || [ -n "${GITHUB_TOKEN-}" ]; then
+  pip config --user set global.disable-pip-version-check True
```
Comment (simon-mo) [inline ci/travis/install-dependencies.sh]: nice!

File: ci/travis/install-dependencies.sh (hunk 4 above)
Comment (mehrdadn) [reply ci/travis/install-dependencies.sh]: Done!

**Review by simon-mo — APPROVED** [review]

**Pull request merged**